File: dotnet/DesignPatterns/CreationalDesignPatterns/FactoryDesignPattern/FactoryDesignPattern.Example1/Concretes/PushNotificationService.cs
using FactoryDesignPattern.NotificationAPI.Abstracts;
using FactoryDesignPattern.NotificationAPI.Models;

namespace FactoryDesignPattern.NotificationAPI.Concretes;

public class PushNotificationService : INotificationService
{
	public string Send(string to, string message)
	{
		string result = $"Sending Push Notification to {to} with message: {message}";
		Console.WriteLine(result);

		return result;
	}
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/FactoryDesignPattern/FactoryDesignPattern.Example1/src/services/PushNotificationService.ts
import { INotificationService } from './INotificationService';

export class PushNotificationService implements INotificationService {
  send(to: string, message: string): string {
    const result = `Sending Push Notification to ${to} with message: ${message}`;
    console.log(result);
    return result;
  }
}
